### PR TITLE
feat: cloudhypervisor api client

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:

This change adds a Cloud Hypervisor API client. This will be used in a
later implementation to enable Flintlock to talk to running instances of
Cloud Hypervisor.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Relates #417 

**Special notes for your reviewer**:

This functionality will not be used initially.

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: pkg/cloudhypervisor/client.go [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+
+// Client represents a client for the Cloud Hypervisor API.
+type Client interface {
+	// VmmPing checks for API server availability.
+	VmmPing(ctx context.Context) (*VmmPingResponse, error)
+	// VmmShutdown shuts down the cloud-hypervisor vmm.
+	VmmShutdown(ctx context.Context) error
+	// Info returns general information about the cloud-hypervisor Virtual Machine (VM) instance.
+	Info(ctx context.Context) (*VmInfo, error)
+	//Counters(ctx context.Context) *VmmCo
```
Comment (Callisto13) [inline pkg/cloudhypervisor/client.go:23]: what dis?

Comment (richardcase) [reply pkg/cloudhypervisor/client.go]: Ah, thats a future todo that i forgot to remove. I have removed it now :)

File: pkg/cloudhypervisor/client.go [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+	// Pause a previously booted VM instance.
+	Pause(ctx context.Context) error
+	// Resume a previously paused VM instance.
+	Resume(ctx context.Context) error
+	// Shutdown the VM instance.
+	Shutdown(ctx context.Context) error
+	// Reboot the VM instance.
+	Reboot(ctx context.Context) error
+	// PowerButton triggers a power button in the VM.
+	PowerButton(ctx context.Context) error
```
[the remark this answers is not in the document]
Comment (richardcase) [reply pkg/cloudhypervisor/client.go]: Well depends on what the power button is configured to do.

[the remark this answers is not in the document]
Comment (richardcase) [reply pkg/cloudhypervisor/client.go]: I updated the comment,

File: pkg/cloudhypervisor/handlers.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		}
+
+		for errorCode, errorMessage := range failureMapping {
+			if errorCode == res.StatusCode {
+				return errors.New(errorMessage)
+			}
+		}
+
+		if res.ContentLength > 0 {
+			data, _ := io.ReadAll(res.Body)
```
[the remark this answers is not in the document]
Comment (richardcase) [reply pkg/cloudhypervisor/handlers.go:27]: Only as we include it in the error message it its greater than 0.

File: pkg/cloudhypervisor/client.go [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+}
+
+// Boot will boot the previously created VM instance.
+func (c *client) Boot(ctx context.Context) error {
+	return c.
+		builder.
+		Clone().
+		Path(PathVmBoot).
+		AddValidator(CustomErrValidator(map[int]string{
+			404: "The VM instance could not boot because it is not created yet",
```
Comment (Callisto13) [inline pkg/cloudhypervisor/client.go:161]: would be good to have all these as consts, but happy to leave that as a TODO for later

File: pkg/cloudhypervisor/handlers.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+		}
+
+		data, err := io.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
+		if err = json.Unmarshal(data, dest); err != nil {
+			return err
+		}
+		return nil
```
Comment (Callisto13) [inline pkg/cloudhypervisor/handlers.go:53]: could be 
```suggestion
		return json.Unmarshal(data, dest)
```
but not bothered

**Pull request merged**